authors/apps/authentication/social.py

```diff
@@ -1,33 +1,41 @@
+'''
+This module contains handlers for social authentication
+'''
 import facebook
-import os
 from google.auth.transport import requests
 from google.oauth2 import id_token
 from django.contrib.auth import authenticate
 from .models import User
 
 
 def register_social_user(email, username):
+    '''
+    This function handles user registration
+    and user login
+    '''
     user = User.objects.filter(email=email)
+    password = User.objects.make_random_password(length=20)
 
     if user.exists():
         current_user = User.objects.get(email=email)
```

Comment:
@JamesMudidi, on registration, if the user already exists simply get the token from the user object.
```
token = current_user.token
return token
```
Replace lines 20 to 25